python/ml4ir/applications/ranking/tests/test_t_test_calculation.py

```diff
@@ -0,0 +1,119 @@
+import numpy as np
+from scipy import stats
+import unittest
+import warnings
+from ml4ir.applications.ranking.model.ranking_model import perform_click_rank_dist_paired_t_test, compute_stats_from_stream
+
+warnings.filterwarnings("ignore")
+
+
+class TestTtestCalculation(unittest.TestCase):
+    """
+    Testing the calculation of paired t-test statistic and its p-value.
+    """
+
+    def t_test_calculation(self, a_bucket, b_bucket, batch_size):
+        n = len(a_bucket)
+        expected_t_test_stat, expected_pvalue = stats.ttest_rel(a_bucket, b_bucket)
+        agg_mean, agg_count, agg_M2 = 0, 0, 0
+        for bcount in range(int(n/batch_size)):
+            batch_a = a_bucket[bcount*batch_size : (bcount+1)*(batch_size)]
+            batch_b = b_bucket[bcount*batch_size : (bcount+1)*(batch_size)]
+            d = batch_a - batch_b
+            agg_count, agg_mean, agg_M2 = compute_stats_from_stream(d, agg_count, agg_mean, agg_M2)
+
+        t_test_stat, pvalue = perform_click_rank_dist_paired_t_test(agg_mean, (agg_M2/(agg_count-1)), agg_count)
+        assert np.isclose(expected_pvalue, pvalue, atol=0.0001)
+
+    def test_1(self):
```

Comment:
@mohazahran This is great. I would also add a unit test for the two methods individually.

Reply:
ping